include/NovelRT/Ecs/DefaultComponentTypes.h

```diff
@@ -0,0 +1,85 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#ifndef NOVELRT_DEFAULTCOMPONENTTYPES_H
+#define NOVELRT_DEFAULTCOMPONENTTYPES_H
+
+#ifndef NOVELRT_ECS_H
+#error NovelRT does not support including types explicitly by default. Please include Ecs.h instead for the Ecs namespace subset.
+#endif
+
+namespace NovelRT::Ecs
+{
+    struct TransformComponent
+    {
+        Maths::GeoVector3F positionAndLayer = Maths::GeoVector3F::zero();
+        Maths::GeoVector2F scale = Maths::GeoVector2F::one();
+        float rotationInEulerAngles = 0.0f;
+
+        inline TransformComponent& operator+=(const TransformComponent& other)
+        {
+            positionAndLayer += other.positionAndLayer;
+            rotationInEulerAngles += other.rotationInEulerAngles;
+            scale += other.scale;
+
+            if (rotationInEulerAngles > 359.999999f) // 360 == 0 pretty sure :]
```

Comment:
I'd probably do this as `>= 360.0f`. `360.0` is exactly representable and will avoid a number of potential precision bugs
This also, notably, isn't accounting for a user passing in something that is say `759.0f`
And some users may expect this to be in radians instead; which also helps avoid precision problems and is a more natural to use representation for internal usage; even if users want to express things in degrees.

Reply:
If we do this in radians we will have to change the internal logic of NovelRT::Maths to use radians also. I did what a game developer would expect of a public API - which is euler angles. If we want to use radians everywhere that's a discussion for another GH issue I think.

I'm happy to change the value though. :)

Reply:
I'm trying to say that there is a disconnect between what's good for internal usage vs what's good for user interface.

Users want to work with degrees; where-as math APIs and most optimizations here will work with `radians`.

Reply:
There's certainly a disconnect in my brain after reading that - this component is public API. What are you wanting me to do here?

Reply:
Pardon the C#, but:
```csharp
private float _rotationInRadians;

public float RotationInDegrees => rad2deg(_rotationInRadians);

public float RotationInRadians => _rotationInRadians;
```

Reply:
....are we sure that won't break the trivially copyable requirement?

Reply:
The term `Euler Angle` here is also confusing, because Euler angles can be specified in degrees or radians. You're specifically expressing `rotation in degrees`

Reply:
I'm using Unity terminology, its what I know. You'll have to bear with me.

Reply:
> ....are we sure that won't break the trivially copyable requirement?

Why would it? You're exposing a getter that does simple arithmetic on value and one that just returns the underlying value,

Reply:
I don't know - its C++ :^)

Reply:
That is, `is_trivially_copyable` doesn't have anything to do whether or not arbitrary members are exposed. It only impacts things that have move/copy constructors and move/copy assignment operators.

Reply:
Gotcha.

Reply:
https://en.cppreference.com/w/cpp/types/is_trivially_copyable

Reply:
I've not implemented the helper members just yet, but wanted to flag that I've changed this functionally to use radians and so I need a second look at it @tannergooding